FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/IncomingMessageInfo.kt

```diff
@@ -84,6 +91,140 @@ data class IncomingMessageInfo constructor(
     return hasSomePart(MsgBlock.Type.PLAIN_TEXT)
   }
 
+  fun toInitializationData(
+    context: Context,
+    @MessageType messageType: Int,
+    accountEmail: String,
+    aliases: List<String>
+  ): InitializationData {
+    val toAddresses: java.util.ArrayList<String> = arrayListOf()
+    val ccAddresses: java.util.ArrayList<String> = arrayListOf()
+    val bccAddresses: java.util.ArrayList<String> = arrayListOf()
+    var body: String? = null
+
+    val folderType = FoldersManager.getFolderType(localFolder)
+    when (messageType) {
+      MessageType.REPLY -> {
+        when (folderType) {
+          FoldersManager.FolderType.SENT,
+          FoldersManager.FolderType.OUTBOX -> {
+            toAddresses.addAll(getTo().map { it.address.lowercase() })
+          }
+
+          else -> {
+            toAddresses.addAll(getReplyToWithoutOwnerAddress().ifEmpty { getTo() }
+              .map { it.address.lowercase() })
+          }
+        }
+      }
+
+      MessageType.REPLY_ALL -> {
+        when (folderType) {
+          FoldersManager.FolderType.SENT, FoldersManager.FolderType.OUTBOX -> {
+            toAddresses.addAll(getTo().map { it.address.lowercase() })
+            ccAddresses.addAll(getCc().map { it.address.lowercase() })
+          }
+
+          else -> {
+            val toRecipients = getReplyToWithoutOwnerAddress().ifEmpty { getTo() }
+            toAddresses.addAll(toRecipients.map { it.address.lowercase() })
+
+            val ccSet = HashSet<InternetAddress>()
```

Comment:
do we need to preserve order? if so, maybe `LinkedHashSet`?

Reply:
Not sure that is needed. But `LinkedHashSet` looks better.